commit 59b4a52e5aa93a86e66cd3475bc42bb8392a9013
Author: Chris Feist <cfeist@redhat.com>
Date:   Thu Jun 17 14:03:06 2010 -0500

    Moved DEFAULT_CLUSTER_VERSION into Globals object to prevent errors

diff --git a/luci/lib/app_globals.py b/luci/lib/app_globals.py
index 8913d74..5b4c120 100644
--- a/luci/lib/app_globals.py
+++ b/luci/lib/app_globals.py
@@ -15,7 +15,6 @@ from luci.lib.ClusterConf.Totem import Totem
 __all__ = ['Globals']
 
 
-DEFAULT_CLUSTER_VERSION = 3
 
 class Globals(object):
     """Container for objects available throughout the life of the application.
@@ -24,6 +23,7 @@ class Globals(object):
     is available during requests via the 'app_globals' variable.
 
     """
+    DEFAULT_CLUSTER_VERSION = 3
 
     def __init__(self):
         """Prepare some common static functions, constants etc."""

commit e4a9dbf059deed4d697849a6350fdb189ab7a8d6
Author: Chris Feist <cfeist@redhat.com>
Date:   Thu Jun 17 15:24:14 2010 -0500

    Nodes now show up even in the nodes interface even if they can't be contacted

diff --git a/luci/controllers/cluster.py b/luci/controllers/cluster.py
index e2ac444..b355481 100644
--- a/luci/controllers/cluster.py
+++ b/luci/controllers/cluster.py
@@ -17,7 +17,7 @@ from pylons.i18n import ugettext as _
 
 # project specific imports
 from luci.lib.base import BaseController
-from luci.lib.db_helpers import get_cluster_list, get_model_for_cluster, get_status_for_cluster, db_remove_cluster, get_agent_for_cluster
+from luci.lib.db_helpers import get_cluster_list, get_model_for_cluster, get_status_for_cluster, db_remove_cluster, get_agent_for_cluster, get_cluster_db_obj
 import luci.lib.ricci_helpers as rh
 import luci.widget_validators.validate_cluster_prop as vcp
 from luci.widget_validators.validate_create_cluster_form import validate_create_cluster_form 
@@ -98,7 +98,9 @@ class IndividualClusterController(BaseController):
 
     @expose("luci.templates.node")
     def nodes(self):
-        return dict(use_referrer=False, page='nodes', name=self.nodename, base_url='/nodes')
+        db = get_cluster_db_obj(self.name)
+
+        return dict(use_referrer=False, page='nodes', name=self.nodename, base_url='/nodes', nodes=db.nodes)
 
     @expose("luci.templates.node")
     def add_nodes_cmd(self, command=None, **kw):
diff --git a/luci/templates/node.html b/luci/templates/node.html
index c45ba4c..a6d1bc8 100644
--- a/luci/templates/node.html
+++ b/luci/templates/node.html
@@ -119,6 +119,23 @@
           <td class="node_tlist_ip">${nodedbobj.hostname}</td>
         </tr>
         </py:if>
+        <py:if test="not cluster_status or not cluster_status.nodes"> 
+        <tr py:for="i in nodes"
+          py:with="hostname = i.hostname">
+          <td class="checkbox"><input type="checkbox" name="${hostname}"/></td>
+          <td class="icon">
+            <img src="${tg.url('/images/question.png')}" alt="Status of the node is unknown." />
+          </td>
+          <td class="main_id">
+            <a href="${tmpl_context.cluster_url + hostname}">
+              <span class="entity_unknown">${hostname}</span>
+            </a>
+          </td>
+          <td class="node_tlist_id"></td>
+          <td class="node_tlist_votes"></td>
+          <td class="node_tlist_status">${_('The status of this node is unknown')}</td>
+        </tr>
+        </py:if>
       </tbody>
     </table>
   </div>
commit 30407ff23dee8de61eecc3e4d127f2115e881c64
Author: Chris Feist <cfeist@redhat.com>
Date:   Fri Jun 18 12:51:42 2010 -0500

    Fixed missing reference to database nodes in controller for node.html

diff --git a/luci/controllers/cluster.py b/luci/controllers/cluster.py
index b355481..5ac310d 100644
--- a/luci/controllers/cluster.py
+++ b/luci/controllers/cluster.py
@@ -94,7 +94,9 @@ class IndividualClusterController(BaseController):
 
     @expose("luci.templates.node")
     def default(self):
-        return dict(page='nodes', name=self.nodename, base_url='/nodes')
+        db = get_cluster_db_obj(self.name)
+
+        return dict(page='nodes', name=self.nodename, base_url='/nodes', nodes=db.nodes)
 
     @expose("luci.templates.node")
     def nodes(self):
@@ -104,7 +106,9 @@ class IndividualClusterController(BaseController):
 
     @expose("luci.templates.node")
     def add_nodes_cmd(self, command=None, **kw):
-        return dict(use_referrer=False, page='nodes', name=self.nodename, base_url='/nodes')
+        db = get_cluster_db_obj(self.name)
+
+        return dict(use_referrer=False, page='nodes', name=self.nodename, base_url='/nodes', nodes=db.nodes)
 
     @expose("luci.templates.node")
     def nodes_fence_cmd(self, command=None, **kw):
